embedded-cassandra/embedded-cassandra-service/src/main/java/org/hawkular/commons/cassandra/EmbeddedCassandraService.java

```diff
@@ -41,22 +41,20 @@
  *
  * @author Stefan Negrea
  */
-@Startup
-@Singleton
+@ApplicationScoped
 public class EmbeddedCassandraService {
 
     private static final Logger logger = LoggerFactory.getLogger(EmbeddedCassandraService.class);
 
-
     private CassandraDaemon cassandraDaemon;
+    private final Object lock = new Object();
 
     public EmbeddedCassandraService() {
         logger.info("======== Hawkular - Embedded Cassandra ========");
     }
 
-    @PostConstruct
-    public void start() {
-        synchronized (this) {
+    public void start(@Observes @Initialized(ApplicationScoped.class) Object ignore) {
```

Comment:
That's a nice trick. Didn't know you could emulate a `@Singleton @Startup` with this.